OMEdit/OMEditGUI/Git/RevertCommitsDialog.cpp

```diff
@@ -0,0 +1,264 @@
+  // Working directory
+  mpWorkingDirectoryLabel = new Label(tr("Working Directory:"));
+  mpWorkingDirectoryTextBox = new QLineEdit;
+  mpWorkingDirectoryTextBox->setText(MainWindow::instance()->getGitCommands()->getRepositoryName(repository));
+  connect(mpWorkingDirectoryTextBox, SIGNAL(textChanged(QString)), this, SLOT(workingDirectoryChanged(QString)));
+  mpDirectoryBrowseButton = new QPushButton(tr("Browse Directory"));
+  connect(mpDirectoryBrowseButton, SIGNAL(clicked()), SLOT(browseWorkingDirectory()));
+  // Commit
+  mpCommitLabel = new Label(tr("Commit:"));
+  mpCommitTextBox = new QLineEdit;
+  mpCommitTextBox->setText("HEAD");
+  mpCommitTextBox->setFocus();
+  mpCommitTextBox->selectAll();
+  mpCommitTextBox->setReadOnly(true);
+  connect(mpCommitTextBox, SIGNAL(textChanged(QString)), this, SLOT(commitTextChanged(QString)));
+  mpCommitBrowseButton = new QPushButton(tr("Browse Commit"));
+  connect(mpCommitBrowseButton, SIGNAL(clicked()), SLOT(browseCommitLog()));
+  mpCommitDescriptionTextBox = new QPlainTextEdit;
+  mpCommitDescriptionTextBox->setReadOnly(true);
+  // Create the buttons
+  mpRevertButton = new QPushButton(tr("Revert"));
+  mpRevertButton->setEnabled(true);
+  connect(mpRevertButton, SIGNAL(clicked()), SLOT(revertCommit()));
+  mpCancelButton = new QPushButton(Helper::cancel);
+  connect(mpCancelButton, SIGNAL(clicked()), SLOT(reject()));
+  // create buttons box
+  mpButtonBox = new QDialogButtonBox(Qt::Horizontal);
+  mpButtonBox->addButton(mpRevertButton, QDialogButtonBox::ActionRole);
+  mpButtonBox->addButton(mpCancelButton, QDialogButtonBox::ActionRole);
+  // set the layout
+  QGridLayout *pMainLayout = new QGridLayout;
+  pMainLayout->setAlignment(Qt::AlignTop | Qt::AlignLeft);
+  pMainLayout->addWidget(mpWorkingDirectoryLabel, 0, 0);
+  pMainLayout->addWidget(mpWorkingDirectoryTextBox, 0, 1);
+  pMainLayout->addWidget(mpDirectoryBrowseButton, 0, 2);
+  pMainLayout->addWidget(mpCommitLabel, 1, 0);
+  pMainLayout->addWidget(mpCommitTextBox, 1, 1);
+  pMainLayout->addWidget(mpCommitBrowseButton, 1, 2);
+  pMainLayout->addWidget(mpCommitDescriptionTextBox, 2, 0, 1, 3);
+  pMainLayout->addWidget(mpButtonBox, 3, 0, 3, 3,  Qt::AlignRight);
+  setLayout(pMainLayout);
+
+  getCommitHistory();
+}
+/*!
+  Slot activated when mpCommitTextBox textChanged signal is raised.\n
+  */
+void RevertCommitsDialog::commitTextChanged(const QString &commit)
+{
+  mpProcess = new QProcess(this);
+  QString repository = MainWindow::instance()->getModelWidgetContainer()->getCurrentModelWidget()->getLibraryTreeItem()->getFileName();
+  QFileInfo fileInfo(repository);
+  QString directory = fileInfo.absoluteDir().absolutePath();
+  mpProcess->setWorkingDirectory(directory);
+  mpProcess->start("git", { "show", "--stat=80", commit });
+  mpProcess->waitForFinished();
+  QString commitHistory =  mpProcess->readAllStandardOutput();
+  mpCommitDescriptionTextBox->setPlainText(commitHistory);
+}
```

Comment:
All the git commands should be done via `gitCommands` class. You also need to provide some settings in the `OptionsDialog` to set the git path. Not all users have git in their PATH.